ops/pebble.py

```diff
@@ -308,6 +317,82 @@ def __repr__(self):
                 ).format(self=self)
 
 
+class Config:
+    """Represents a Pebble config layer (or flattened configuration).
+
+    This is not documented, but captured in code here:
+    https://github.com/canonical/pebble/blob/master/internal/setup/setup.go
+    """
+
+    def __init__(self, raw: Union[str, Dict]):
+        if isinstance(raw, str):
+            d = yaml.safe_load(raw)
+        else:
+            d = raw
+        self.summary = d.get('summary', '')
+        self.description = d.get('description', '')
+        self.services = {name: ConfigService(name, service)
+                         for name, service in d.get('services', {}).items()}
+
+    def to_yaml(self) -> str:
+        """Convert this config to its YAML representation."""
+        return yaml.dump(self.to_dict(), Dumper=_DefaultDumper)
```

Comment:
DefaultDumper probably needs a better name. It's not clear why it is being provided (safety), and it's not actually the default since we need to provide it explicitly everywhere.

Reply:
Good call, changed to `_safe_dumper`.